0.6 updates

dwfmarchant opened the pull request:
I think this is ready to merge.  @Pbellive ?

Pbellive committed:
Updated JOcTree to remove deprecation warnings in 0.6.

Pbellive committed:
Added storage of transpose of edge mass matrix integration matrix and other small changes to getdEdgeMassMatrix to speed up formation of dM during sensitivity computations.

dwfmarchant committed:
Make getdEdgeMassMatrix work again for complex v.

dwfmarchant committed:
Remove createOctree codes that no longer work.

dwfmarchant committed:
Added getCellCenterGradientMatrix and getSizeOfNeighbors.

dwfmarchant committed:
Added outputOctreeMesh.  Bug fixes.

Pbellive commented:
I forget again if we made a decision about maintaining 0.5 compatibility. Both Travis and Appveyor are running tests on 0.5 and they're failing. I just looked into backward compatibility and I think it should be doable with an hour or so's work. I think we should either update the CI settings and require Julia 0.6 or add 0.5 compatibility, which I can do now---basically means adding `@compat` to the front of a few lines.

dwfmarchant commented:
I'd be fine with just dropping 0.5 support.  @lruthotto ?

dwfmarchant commented:
I talked with Eldad and Christoph over here and they are fine with just dropping 0.5 support.

dwfmarchant committed:
Updates to CI configs.

dwfmarchant committed:
Update readme and require to 0.6

Pbellive commented:
Re dropping 0.5 support: Sounds good. Looks like we're ready to go. I'll merge this once https://github.com/JuliaInv/jInv.jl/pull/45 is merged.

dwfmarchant committed:
Fix warnings in getInterpolationMatrix in 0.6.

Commit:
Fix more warnings for 0.6

findBlocks.jl
getNodalInterpolationMatrix.jl

Commit:
Fix getNodalConstraints

The last return argument of getNodalConstraints was not returning the
lookup table for the nodal numbers on the constrained OcTree.
